chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/SendEmailJob.java

```diff
@@ -0,0 +1,224 @@
+        properties.put("mail.smtp.auth", "true");
+        properties.put("mail.smtp.starttls.enable", "true");
+        properties.put("smtpUsername", env.getProperty("smtpUsername"));
+        properties.put("smtpPassword", env.getProperty("smtpPassword"));
+        properties.put("smtpFrom", env.getProperty("smtpFrom"));
+
+        return properties;
+    }
+
+    private MimeMessage getMimeMessage(ChplEmailMessage message) throws MessagingException {
+
+        EmailOverrider overrider = new EmailOverrider(env);
+        Session session = Session.getInstance(getProperties(), getAuthenticator(getProperties()));
+
+        MimeMessage mimeMessage = new MimeMessage(session);
+        mimeMessage.addRecipients(RecipientType.TO, overrider.getRecipients(message.getRecipients()));
+        mimeMessage.setFrom(new InternetAddress(getProperties().getProperty("smtpFrom")));
+        mimeMessage.setSubject(message.getSubject());
+        mimeMessage.setSentDate(new Date());
+
+        Multipart multipart = new MimeMultipart();
+        multipart.addBodyPart(overrider.getBody(message.getBody(), message.getRecipients()));
+        if (message.getFileAttachments() != null && message.getFileAttachments().size() > 0) {
+            // Add file attachments to email
+            for (File file : message.getFileAttachments()) {
+                MimeBodyPart messageBodyPart = new MimeBodyPart();
+                DataSource source = new FileDataSource(file);
+                messageBodyPart.setDataHandler(new DataHandler(source));
+                messageBodyPart.setFileName(file.getName());
+                multipart.addBodyPart(messageBodyPart);
+            }
+        }
+        mimeMessage.setContent(multipart, "text/html; charset=UTF-8");
+        return mimeMessage;
+    }
+
+    private Date getRetryTime() {
+        Integer retryInterval = Integer.valueOf(env.getProperty("emailRetryIntervalInMinutes"));
+        Calendar cal = Calendar.getInstance();
+        cal.add(Calendar.MINUTE, retryInterval);
+        return cal.getTime();
+    }
+
+    private Integer getMaxRetryAttempts() {
+        return Integer.valueOf(env.getProperty("emailRetryAttempts"));
+    }
+
+    private List<File> copyFilesOnFirstReschedule(ChplEmailMessage message) {
+        if (message.getRetryAttempts().equals(1) && message.getFileAttachments() != null) {
+            if (message.getFileAttachments() == null || message.getFileAttachments().size() == 0) {
+                LOGGER.info("No files to move.");
+            }
+            return  message.getFileAttachments().stream()
+                    .map(file -> copyTempFileToPermanentLocation(file))
+                    .collect(Collectors.toList());
+        } else {
+            return message.getFileAttachments();
+        }
+
+    }
+
+    private File copyTempFileToPermanentLocation(File originalFile) {
+        Path newPath = Paths.get(env.getProperty("downloadFolderPath") + File.separator + originalFile.getName());
```

Comment:
Is this the directory we want to use? Or would we rather have a new and distinct folder we add to the machines that we use for this?